### ISSUE TITLE
It's taking FOREVER! is this a dedrm issue?

### ISSUE THREAD

**Issue by AboShayeb** [apprenticeharper/DeDRM_tools#1449]
Convert book 1 of 1 (Biomedical Applications of Introductory Physics)
Conversion options changed from defaults:
  read_metadata_from_opf: u'C:\\Users\\ISMAEE~1\\AppData\\Local\\Temp\\calibre_1frled\\v3d7zz.opf'
  output_profile: u'generic_eink'
  cover: u'C:\\Users\\ISMAEE~1\\AppData\\Local\\Temp\\calibre_1frled\\c9j38q.jpeg'
  verbose: 2
  pdf_serif_family: u'MS Shell Dlg 2'
  pdf_sans_family: u'MS Shell Dlg 2'
Resolved conversion options
calibre version: 4.23.0
{'asciiize': False,
 'author_sort': None,
 'authors': None,
 'base_font_size': 0.0,
 'book_producer': None,
 'change_justification': u'original',
 'chapter': u"//*[((name()='h1' or name()='h2') and re:test(., '\\s*((chapter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', 'i')) or @class = 'chapter']",
 'chapter_mark': u'pagebreak',
 'comments': None,
 'cover': u'C:\\Users\\ISMAEE~1\\AppData\\Local\\Temp\\calibre_1frled\\c9j38q.jpeg',
 'custom_size': None,
 'debug_pipeline': None,
 'dehyphenate': True,
 'delete_blank_paragraphs': True,
 'disable_font_rescaling': False,
 'duplicate_links_in_toc': False,
 'embed_all_fonts': False,
 'embed_font_family': None,
 'enable_heuristics': False,
 'expand_css': False,
 'extra_css': None,
 'filter_css': u'',
 'fix_indents': True,
 'font_size_mapping': None,
 'format_scene_breaks': True,
 'html_unwrap_factor': 0.4,
 'input_encoding': None,
 'input_profile': <calibre.customize.profiles.InputProfile object at 0x000001AAF252EF48>,
 'insert_blank_line': False,
 'insert_blank_line_size': 0.5,
 'insert_metadata': False,
 'isbn': None,
 'italicize_common_cases': True,
 'keep_ligatures': False,
 'language': None,
 'level1_toc': None,
 'level2_toc': None,
 'level3_toc': None,
 'line_height': 0.0,
 'linearize_tables': False,
 'margin_bottom': 5.0,
 'margin_left': 5.0,
 'margin_right': 5.0,
 'margin_top': 5.0,
 'markup_chapter_headings': True,
 'max_toc_links': 50,
 'minimum_line_height': 120.0,
 'new_pdf_engine': False,
 'no_chapters_in_toc': False,
 'no_images': False,
 'no_inline_navbars': False,
 'output_profile': <calibre.customize.profiles.GenericEink object at 0x000001AAF253EB88>,
 'page_breaks_before': u"//*[name()='h1' or name()='h2']",
 'paper_size': u'letter',
 'pdf_add_toc': False,
 'pdf_default_font_size': 20,
 'pdf_footer_template': None,
 'pdf_header_template': None,
 'pdf_hyphenate': False,
 'pdf_mark_links': False,
 'pdf_mono_family': u'Courier',
 'pdf_mono_font_size': 16,
 'pdf_odd_even_offset': 0.0,
 'pdf_page_margin_bottom': 72.0,
 'pdf_page_margin_left': 72.0,
 'pdf_page_margin_right': 72.0,
 'pdf_page_margin_top': 72.0,
 'pdf_page_number_map': None,
 'pdf_page_numbers': False,
 'pdf_sans_family': u'MS Shell Dlg 2',
 'pdf_serif_family': u'MS Shell Dlg 2',
 'pdf_standard_font': u'serif',
 'pdf_use_document_margins': False,
 'prefer_metadata_cover': False,
 'preserve_cover_aspect_ratio': False,
 'pretty_print': False,
 'pubdate': None,
 'publisher': None,
 'rating': None,
 'read_metadata_from_opf': u'C:\\Users\\ISMAEE~1\\AppData\\Local\\Temp\\calibre_1frled\\v3d7zz.opf',
 'remove_fake_margins': True,
 'remove_first_image': False,
 'remove_paragraph_spacing': False,
 'remove_paragraph_spacing_indent_size': 1.5,
 'renumber_headings': True,
 'replace_scene_breaks': u'',
 'search_replace': '[]',
 'series': None,
 'series_index': None,
 'smarten_punctuation': False,
 'sr1_replace': None,
 'sr1_search': None,
 'sr2_replace': None,
 'sr2_search': None,
 'sr3_replace': None,
 'sr3_search': None,
 'start_reading_at': None,
 'subset_embedded_fonts': False,
 'tags': None,
 'timestamp': None,
 'title': None,
 'title_sort': None,
 'toc_filter': None,
 'toc_threshold': 6,
 'toc_title': None,
 'transform_css_rules': '[]',
 'uncompressed_pdf': False,
 'unit': u'inch',
 'unsmarten_punctuation': False,
 'unwrap_factor': 0.45,
 'unwrap_lines': True,
 'use_auto_toc': False,
 'use_profile_size': False,
 'verbose': 2}
InputFormatPlugin: PDF Input running
on C:\Users\ISMAEE~1\AppData\Local\Temp\calibre_1frled\8ukvwd.pdf
Converting file to html...

**Comment by ElleKayEm** [comment]
Hard to tell anything from this log, as opposed to an import debug log.  Unless it errors out saying that it can't convert because of DRM, it's probably a just large file taking a long time to convert.  If you had DeDRM installed before adding the book to calibre, then DRM should have already been removed when you first added the book.

**Comment by AboShayeb** [comment]
Dude wait!!! Are you saying that I only have to import the book in and then copy the file from calibre library folder? Like I don't have to convert it whatsoever?
Please can you tell me the shortest way i can dedrm a pdf book off ade?

**Comment by ElleKayEm** [comment]
Yes, if everything is working correctly, DRM removal happens when you add the book to calibre.

**Comment by lnxme** [comment]
@AboShayeb
In Calibre, look in the "Preferences" menu  then "Adding Books" then "Adding Actions" and unclick the box "Automatically convert added books to the preferred output format" and apply the changes.
Now when you add a book it will just add it (removing any DRM)  and will not attempt to convert it.
